http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -1262,6 +1303,65 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
     _ <- if (res.isEmpty) authProvider.checkUserEnabled(userInfo) else F.unit
   } yield res
 
+  private def filterAppsBySamProjectPermission(allClusters: List[KubernetesCluster],
```

Comment:
I am not 100% sure, but I believe the logic of partitioning apps by access scope already exists somewhere (Rob added it in), would it make sense to reuse it?

Reply:
yeah I tried to reuse that function (`filterAppsBySamPermission`), but there's enough changes required in multiple parts of the function that it required some funky logic to not break it out (still need the original function for v2 endpoints), I can take another stab at it though

Reply:
Nah, no need, just wanted to make sure you saw the original logic, if it is different enough, no need to refactor